pkg/report/reporter.go

```diff
@@ -140,37 +140,51 @@ func (d *defaultReporter) updateSummaryFromRecord(r Record) {
 
 	d.ended = time.Time(r.Time)
 	switch r.State {
-	case StateDeploySuccess:
+	case StateSuccess:
 		d.deploymentsSuccessCount++
-	case StateDeployExcluded:
+	case StateExcluded:
 		d.deploymentsExcludedCount++
-	case StateDeploySkipped:
+	case StateSkipped:
 		d.deploymentsSkippedCount++
-	case StateDeployError:
+	case StateError:
 		d.deploymentsErrorCount++
 	default:
 		panic(fmt.Sprintf("unexpected state for deployment event: %s", r.State))
 	}
 }
 
 // ReportDeployment reports the result of deploying a config.
-func (d *defaultReporter) ReportDeployment(config coordinate.Coordinate, state string, details []Detail, err error) {
-	d.queue <- Record{
+func (d *defaultReporter) ReportDeployment(config coordinate.Coordinate, state RecordState, details []Detail, err error) {
+	record := Record{
 		Type:    TypeDeploy,
 		Time:    JSONTime(d.clockFunc()),
-		Config:  config,
+		Config:  &config,
 		State:   state,
 		Details: details,
 		Error:   convertErrorToString(err),
 	}
+
+	d.updateSummaryFromRecord(record)
+	d.queue <- record
 }
 
-func convertErrorToString(err error) *string {
+// ReportLoading reports the result of validating a config (manifest, project, config).
+func (d *defaultReporter) ReportLoading(state RecordState, err error, message string, config *coordinate.Coordinate) {
```

Comment:
No dedicated test coverage

Reply:
On the other side, it's tested via `loading-report.test.go`